Fix some errors and add some default parameters

viagram opened the pull request.

mosajjal commented:
hey @viagram, made some changes based on these suggestions. Let me know if it addresses issues you've raise in this PR. Since a lot of people disable `systemd-resolved` to get this working, I introduced a secondary method to get the public IP through DNS. I also overhauled the whole flag system with POSIX ones which support short flags (-c).

viagram commented:
> hey @viagram, made some changes based on these suggestions. Let me know if it addresses issues you've raise in this PR. Since a lot of people disable `systemd-resolved` to get this working, I introduced a secondary method to get the public IP through DNS. I also overhauled the whole flag system with POSIX ones which support short flags (-c).

👍👍👍 Very nice
